app/femr/data/models/mysql/User.java

```diff
@@ -55,6 +55,10 @@
     private Boolean passwordReset;
     @Column(name = "notes")
     private String notes;
+    @Column (name = "creation_date", nullable = false) //Sam Zanni
+    private DateTime DateCreated; //Sam Zanni
+    @Column (name = "user_created", nullable = false) //Sam Zanni
+    private String UserCreated; //Sam Zanni
```

Comment:
Hey Ken,

I pushed some more changes for the trackable columns in the same way that I
did the new changes for the weeksPregnant. I tried doing the evolution
within the migration the way that you sent it over but had several errors
and would not work, therefore, I kept the old migration that I had and
switched the variable names. If you could, could you please look this over?
Thanks! If I can't figure out the new migration, would it be possible to go
over this on Tuesday at the meeting? Thank you and have a good rest of the
day!

Sincerely,
Samantha

On Sun, Nov 15, 2015 at 3:34 PM, Samantha Zanni <zanni022@gmail.com> wrote:

> Hey Ken!
>
> Okay, thanks! I will start making those changes and I will also wait to
> hear Kevin's input before I finalize the changes. Thanks again!
>
> Sincerely,
> Samantha
>
> On Sun, Nov 15, 2015 at 2:48 PM, Ken Dunlap <notifications@github.com>
> wrote:
>
>> In app/femr/data/models/mysql/User.java
>> <https://github.com/FEMR/femr/pull/445#discussion_r44876597>:
>>
>> > @@ -55,6 +55,10 @@
>> >      private Boolean passwordReset;
>> >      @Column(name = "notes")
>> >      private String notes;
>> > +    @Column (name = "creation_date", nullable = false) //Sam Zanni
>> > +    private DateTime DateCreated; //Sam Zanni
>> > +    @Column (name = "user_created", nullable = false) //Sam Zanni
>> > +    private String UserCreated; //Sam Zanni
>>
>> I think that UserCreated should link to an actual user_id rather than a
>> string of the first and last name. I also think UserCreated is kind of
>> ambiguous, maybe change it to CreatedBy?
>>
>> private User CreatedBy;
>>
>> This change would affect the migrations as well. The user_created column
>> in the migration would be something like:
>>
>> ADD COLUMN `created_by` INT NOT NULL AFTER `creation_date`;
>> CONSTRAINT `fk_users_created_by_users_id`
>>     FOREIGN KEY (`user_id`)
>>     REFERENCES `users` (`id`));
>>
>> Not sure if this is 100% correct. The downside is that we'd be adding a
>> self-referencing column to the users table. I know we've had some weird
>> issues with this in the past in regards to prescriptions and replacements.
>> We might want to talk about this @kevinzurek
>> <https://github.com/kevinzurek>: do you have any input here?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/FEMR/femr/pull/445/files#r44876597>.
>>
>
>